modin/config/envvars.py

```diff
@@ -524,6 +524,8 @@ class HdkLaunchParameters(EnvironmentVariable, type=dict):
         "null_div_by_zero": 1,
         "enable_watchdog": 0,
         "enable_thrift_logs": 0,
+        "enable_lazy_dict_materialization": 0,
```

Comment:
What about something like this?

```python
    @classmethod
    def _get_default(cls) -> Any:
        if (default := getattr(cls, "default", None)) is None:
            import pyhdk
            from packaging import version

            cls.default = default = {
                "enable_union": 1,
                "enable_columnar_output": 1,
                "enable_lazy_fetch": 0,
                "null_div_by_zero": 1,
                "enable_watchdog": 0,
                "enable_thrift_logs": 0,
            }

            if version.parse(pyhdk.__version__) >= version.parse("0.6.1"):
                default["enable_lazy_dict_materialization"] = 0
                default["log_dir"] = "pyhdk_log"
        return default
```

Reply:
Sure, I can make that change. Would we still need to check here, though? https://github.com/modin-project/modin/pull/6303/files#diff-229f59982e185af8b8f88540fae04f56bbb400295a6116bad2feff4ea5ba1f9cR44 
If for some reason a user set the feature flag `log_dir` but had an older version of HDK, the command line parse could fail.

Reply:
I think we do not need to check here. If a user set the `log_dir` option, it's better to fail with a corresponding error message, rather than removing the unsupported option silently.